### PR TITLE
Select term page

### PR DESCRIPTION
**Pull request by ngw**
## Ticket and context

Ticket: https://dfedigital.atlassian.net/browse/CST-101

### CODE REVIEW

File: app/views/schools/add_participants/start_term.html.erb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    <%= form_for add_participant_form, url: { action: :update }, method: :patch do |f| %>
+
+      <%= f.govuk_radio_buttons_fieldset :start_term,
+        legend: { text: "When do you expect #{add_participant_form.full_name} to start their induction programme?", tag: 'h1', size: 'xl' } do %>
+
+        <% ParticipantProfile.start_terms.each do |term_key, term_value| %>
+          <%= f.govuk_radio_button :start_term, term_key, label: { text: term_value } %>
+        <% end %>
+      <% end %>
```
Comment (ethanmills) [inline app/views/schools/add_participants/start_term.html.erb:17]: Have you looked at `govuk_collection_radio_buttons`? This is what it was designed for https://govuk-form-builder.netlify.app/form-elements/radios/

**Review by ethanmills — APPROVED** [review]

**Comment by ethanmills** [comment]
blocked on clarification of requirements. Meeting in for tomorrow morning to discuss

File: app/forms/schools/add_participant_form.rb [hunk 2]
```diff
@@ -110,6 +123,14 @@ def current_user
       @current_user ||= User.find_by(id: current_user_id)
     end
 
+    def legend
+      if mentor_options.any?
+        "When do you expect #{full_name} to begin mentoring ECTs?"
+      else
+        "When do you expect #{full_name} to start their induction programme?"
+      end
+    end
```
Comment (ethanmills) [inline app/forms/schools/add_participant_form.rb:132]: There's probably a cleaner way to do this

File: app/views/schools/participants/edit_start_term.html.erb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<% content_for :before_content, govuk_back_link( text: "Back", href: schools_participant_path(id: @profile)) %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <%= form_for @profile, url: schools_participant_update_start_term_path, method: :put do |f| %>
+      <%= f.govuk_radio_buttons_fieldset :start_term,
+          legend: { text: "Change start induction", tag: 'h1', size: 'xl' } do %>
```
Comment (ethanmills) [inline app/views/schools/participants/edit_start_term.html.erb:7]: the copy here will probably need changing - maybe ask a content designer?

**Comment by BroiSatse** [comment]
Look all good to me. The only thing that worries me are the hardcoded options to select from (will we need to redploy in january to add/remove other options?) and the format of start_term we store in the database - it's a bit too human friendly for my liking, would prefer sth like `[:summer, 2020]`.

**Review by BroiSatse — APPROVED** [review]

**Pull request merged**